rfd/0086/config.md

```diff
@@ -2,174 +2,236 @@
-
-  consul-agent:
-    port: 8500
-    command: consul -agent -join consul
-    advertise: false
-    restart: always
-
-  consul-template:
-    command: >
-      consul-template -consul consul
-          -template /tmp/template.ctmpl:/tmp/result
-    advertise: false
-    restart: always
-
-  task1:
-    command: /usr/local/bin/tash.sh arg1
-    frequency: 1500ms
-    timeout: 100ms
-    advertise: false
-
-
-health:
-  nginx:
-    check-A:
-      command: >
-        /usr/bin/curl --fail -s -o /dev/null http://localhost/app
+```json5
+{
+  consul: {
+    host: "localhost:8500"
+  },
+  logging: {
+    level: "INFO",
+    format: "default",
+    output: "stdout"
+  },
+  jobs: [
+    {
+      name: "app",
+      // this is upstart-like syntax indicating we want to start this
+      // service when the "setup" service has exited with success but
+      // give up after 60 sec
+      when: "setup exitSuccess timeout 60s",
+      exec: "/bin/app",
+      restart: "never",
+      port: 80,
+      heartbeat: 5,
+      tll: 10,
+      stopTimeout: 5,
+      tags: [
+        "app",
+        "prod"
+      ],
+      interfaces: [
+        "eth0",
+        "eth1[1]",
+        "192.168.0.0/16",
+        "2001:db8::/64",
+        "eth2:inet",
+        "eth2:inet6",
+        "inet",
+        "inet6",
+        "static:192.168.1.100", // a trailing comma isn't an error!
+        ]
+    },
+    {
+      name: "setup",
+      // we can create a chain of "prestart" events
+      when: "consul-agent healthy",
+      exec: "/usr/local/bin/preStart-script.sh",
+      restart: "never"
+    },
+    {
+      name: "preStop",
+      when: "app stopping",
+      exec: "/usr/local/bin/preStop-script.sh",
+      restart: "never",
+    },
+    {
+      name: "postStop",
+      when: "app stopped",
+      exec: "/usr/local/bin/postStop-script.sh",
+    },
+    {
+      // a service that doesn't have a "when" field starts up on the
+      // global "startup" event by default
+      name: "consul-agent",
+      // note we don't have a port here because we don't intend to
+      // advertise one to the service discovery backend
+      exec: "consul -agent -join consul",
+      restart: "always"
+    },
+    {
+      name: "consul-template",
+      exec: ["consul-template", "-consul", "consul",
+             "-template", "/tmp/template.ctmpl:/tmp/result"],
+      restart: "always",
+    },
+    {
+      name: "task1",
+      exec: "/usr/local/bin/tash.sh arg1",
+      frequency: "1500ms",
+      timeout: "100ms",
+    },
+    {
+      name: "reload-app",
+      when: "watch.app changes",
+      exec: "/usr/local/bin/reload-app.sh",
+    },
+    {
+      name: "reload-nginx",
+      when: "watch.nginx changes",
+      exec: "/usr/local/bin/reload-nginx.sh",
+    }
+  ],
+  health: {
+    {
+      name: "checkA",
+      service: "nginx",
+      exec: "/usr/bin/curl --fail -s -o /dev/null http://localhost/app",
+      poll: 5,
+      timeout: "5s",
+    }
+  }
+  watches: {
```

Comment:
Perhaps a comment here explaining that `watches` replace `upstreams` from v2, and they trigger events that can be hooked with `jobs` defined above.

There's a lot of depth in https://github.com/joyent/containerpilot/issues/227, but summarizing it here (which will probably be the basis of our docs) might help clarify the conclusions.